misc/tests/test-dependencies.py

```diff
@@ -1,63 +1,86 @@
 import sys
 
 DIR = os.path.dirname(os.path.abspath(__file__))
 REPO = os.path.dirname(os.path.dirname(DIR))
-DOWNWARD_FILES = os.path.join(REPO, "src", "search", "DownwardFiles.cmake")
-TEST_BUILD_CONFIGS = os.path.join(REPO, "test_build_configs.py")
-BUILD = os.path.join(REPO, "build.py")
-BUILDS = os.path.join(REPO, "builds")
-paths_to_clean = [TEST_BUILD_CONFIGS]
-
-
-def clean_up(paths_to_clean):
-    print("\nCleaning up")
-    for path in paths_to_clean:
-        print("Removing {path}".format(**locals()))
-        if os.path.isfile(path):
-            os.remove(path)
-        if os.path.isdir(path):
-            shutil.rmtree(path)
-    print("Done cleaning")
-
-
-with open(DOWNWARD_FILES) as d:
-    content = d.readlines()
-
-content = [line for line in content if '#' not in line]
-content = [line for line in content if 'NAME' in line or 'CORE_PLUGIN' in line or 'DEPENDENCY_ONLY' in line]
-
-plugins_to_be_tested = []
-for line in content:
-    if 'NAME' in line:
-        plugins_to_be_tested.append(line.replace("NAME", "").strip())
-    if 'CORE_PLUGIN' in line or 'DEPENDENCY_ONLY' in line:
-        plugins_to_be_tested.pop()
-
-with open(TEST_BUILD_CONFIGS, "w") as f:
-    for plugin in plugins_to_be_tested:
-        lowercase = plugin.lower()
-        line = "{lowercase} = [\"-DCMAKE_BUILD_TYPE=Debug\", \"-DDISABLE_PLUGINS_BY_DEFAULT=YES\"," \
-               " \"-DPLUGIN_{plugin}_ENABLED=True\"]\n".format(**locals())
-        f.write(line)
-        paths_to_clean.append(os.path.join(BUILDS, lowercase))
-
-plugins_failed_test = []
-for plugin in plugins_to_be_tested:
+LIBRARY_DEFINITION_FILE = os.path.join(REPO, "src", "search", "DownwardFiles.cmake")
+BUILDS = os.path.join(REPO, "builds/test-dependencies")
+
+
+# Find the closing bracket matching the opening bracket that occurred before start
+def find_corresponding_closing_bracket(content, start):
+    nested_level = 1
+    for index, character in enumerate(content[start:]):
+        if character == "(":
+            nested_level += 1
+        if character == ")":
+            nested_level -= 1
+            if nested_level == 0:
+                return index+start
+
+# Extract all "create_fast_downward_library(...)" blocks and if the library is
+# not a core or depencency library, add its name to the return list.
+def get_library_definitions(content):
+    libraries = []
+    library_definition_string = r"create_fast_downward_library\("
+    pattern = re.compile(library_definition_string)
+    for library_match in pattern.finditer(content):
+        start = library_match.start()+len(library_definition_string)
+        end = find_corresponding_closing_bracket(content, start)
+        library_definition = content[start:end]
+        # we cannot manually en-/disable core and dependency only libraries
+        if any(s in library_definition for s in ["CORE_LIBRARY", "DEPENDENCY_ONLY"]):
+            continue
+        name_match = re.search(r"NAME (\S+)", library_definition)
```

Comment:
To be on the safe side, I'd change `NAME (\S+)` to `NAME\s+(\S+)`, otherwise an extra space breaks the pattern.